src/hooks/useStore.js

```diff
@@ -1,104 +1,111 @@
-			name: 'Squats',
-			image: 'https://images.unsplash.com/photo-1574680178050-55c6a6a96e0a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1469&q=80',
-			description: 'This is an instruction to do your exercise.',
-			sets: 0,
-			reps: 0,
-			weight: 0,
-			isDone: false,
-			isBookmarked: false,
-			isPlanned: false,
-			date: [],
-		},
-		{
-			id: 1,
-			name: 'Biceps Curls',
-			image: 'https://images.unsplash.com/photo-1574680178050-55c6a6a96e0a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1469&q=80',
-			description: 'This is an instruction to do your exercise.',
-			sets: 0,
-			reps: 0,
-			weight: 0,
-			isDone: false,
-			isBookmarked: false,
-			isPlanned: false,
-			date: [],
-		},
-		{
-			id: 2,
-			name: 'Shoulder Press',
-			image: 'https://images.unsplash.com/photo-1574680178050-55c6a6a96e0a?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1469&q=80',
-			description: 'This is an instruction to do your exercise.',
-			sets: 0,
-			reps: 0,
-			weight: 0,
-			isDone: false,
-			isBookmarked: false,
-			isPlanned: false,
-			date: [],
-		},
-	],
+	exerciseCards,
+	setExerciseDate: (id, date) => {
+		set(state => ({
+			exerciseCards: state.exerciseCards.map(exerciseCard =>
+				exerciseCard.id === id ? {...exerciseCard, date} : exerciseCard
+			),
+		}));
+	},
+
+	workouts: [],
 
 	changeSets: (id, sets) => {
 		set(state => ({
-			cards: state.cards.map(card => (card.id === id ? {...card, sets} : card)),
+			workouts: state.workouts.map(workout =>
+				workout.id === id ? {...workout, sets} : workout
+			),
 		}));
 	},
 
 	changeReps: (id, reps) => {
 		set(state => ({
-			cards: state.cards.map(card => (card.id === id ? {...card, reps} : card)),
+			workouts: state.workouts.map(workout =>
+				workout.id === id ? {...workout, reps} : workout
+			),
 		}));
 	},
 
 	changeWeight: (id, weight) => {
 		set(state => ({
-			cards: state.cards.map(card => (card.id === id ? {...card, weight} : card)),
+			workouts: state.workouts.map(workout =>
+				workout.id === id ? {...workout, weight} : workout
+			),
 		}));
 	},
 
-	deleteFromPlanner: id => {
-		set(state => ({
-			cards: state.cards.map(card => (card.id === id ? {...card, isPlanned: false} : card)),
-		}));
+	deleteWorkout: id => {
+		set(state => {
+			return {
+				workouts: state.workouts.filter(workout => id !== workout.id),
+			};
+		});
 	},
 
-	addToPlanner: (id, date) => {
-		set(state => ({
-			cards: state.cards.map(card =>
-				card.id === id ? {...card, isPlanned: true, date: [...card.date, date]} : card
-			),
-		}));
+	addToPlanner: id => {
+		set(state => {
+			return {
+				workouts: [
+					...state.workouts,
+					...state.exerciseCards
+						.filter(exerciseCard => exerciseCard.id === id)
+						.map(exerciseCard =>
+							exerciseCard.id === id ? {...exerciseCard, id: nanoid()} : exerciseCard
+						),
+				],
+			};
+		});
 	},
```

Comment:
Instead of chaining filter and map, you could use the method `.find()`. The `.filter()` method returns an array, whereas `.find()` will return the object itself, so you do not need to use `.map()` afterwards.

```suggestion
addToPlanner: id => {
		set(state => {
			return {
				workouts: [
					...state.workouts,
					{
						...state.exerciseCards.find(exerciseCard => exerciseCard.id === id),
						id: nanoid(),
					},
				],
			};
		});
	},
```